src/store/modules/entity.ts

```diff
@@ -0,0 +1,220 @@
+import { ActionContext } from "vuex";
+import { State as RootState } from "@/store/index";
+import Relationship from "@/services/api/v2/relationship.service";
+import { ITopic } from "@/interfaces/api/v2/topic.interface";
+import { IEntityFacetField } from "@/interfaces/api/v2/entityFilter.interface";
+import { ICredentialSet } from "@/interfaces/api/v2/credential-set.interface";
+import { ICredentialDisplayType } from "@/interfaces/api/v4/credential.interface";
+import { isRegType } from "@/utils/entity";
+
+export type Filter = { [key: string]: string | Array<string> | boolean };
+
+const relationshipService = new Relationship();
+
+export interface State {
+  credentialsExpanded: boolean;
+  relationships: Relationship[];
+  selected: {
+    topic: ITopic | null;
+    credentialSet: ICredentialSet | null;
+  };
+  entityFilter: Filter | null;
+  filterValues:{
+    authorities: Array<IEntityFacetField>;
+    credentialType: Array<IEntityFacetField>;
+    registrationType: Array<IEntityFacetField>;
+  };
+  
+}
+
+const state: State = {
+  credentialsExpanded: false,
```

Comment:
Why are we storing component UI state here in  a store? While I get some may argue that the store should be where all state is kept, I like to think it more useful for component data. It's sometimes less complex to just store UI state like this directly in the parent view component.

Reply:
because the credentialsExpanded has to pass to every credentialItem  in the EntityResults. I could do this with props if you'd like but I just found it simpler to put it in the store